pkg/kncloudevents/message_receiver_test.go

```diff
@@ -170,6 +172,45 @@ func TestStartListenReceiveEvent(t *testing.T) {
 
 }
 
+func TestWithWriteTimeout(t *testing.T) {
+	writeTimeout := time.Millisecond * 10
+	writtenResponse := make(chan bool, 1)
+	receivedRequest := make(chan bool, 1)
+	messageReceiver := NewHTTPMessageReceiver(0, WithWriteTimeout(writeTimeout))
+	ctx := context.TODO()
+
+	go func() {
+		_ = messageReceiver.StartListen(ctx, &blockingHandler{
+			blockFor:        writeTimeout * 10,
+			writtenResponse: writtenResponse,
+			receivedRequest: receivedRequest,
+		})
+	}()
+
+	<-messageReceiver.Ready
+
+	addr := "http://" + messageReceiver.server.Addr
+	client := http.DefaultClient
+	req, err := http.NewRequest("GET", addr, nil)
+	assert.NoError(t, err)
+	_, err = client.Do(req)
+	<- receivedRequest
+
+	assert.Error(t, err)
+	// go http server implementation will abruptly close the connection and the client will only see an io EOF error
+	// due to a buffer.
+	// issue tracking an enhancement: https://github.com/golang/go/issues/21389
+	assert.EqualError(t, err, fmt.Sprintf("Get \"%s\": EOF", addr))
+}
+
+func TestWithReadTimeout(t *testing.T) {
```

Comment:
I realized after writing the tests for the `WriteTimeout` that we only need to test that the configuration is getting plumbed through to net/http server. No need to test that the server is respecting the timeouts as that should be covered by net/http. Can simplify the `TestWithWriteTimeout` as well if needed.